internal/pyxis/submit.go

```diff
@@ -15,6 +16,13 @@ var defaultRegistryAlias = "docker.io"
 func (p *pyxisClient) SubmitResults(ctx context.Context, certInput *CertificationInput) (*CertificationResults, error) {
 	var err error
 
+	// Create the test results, so we can fail fast if version check throws error.
+	testResults := certInput.TestResults
+	testResults, err = p.createOrUpdateTestResults(ctx, testResults, http.MethodPost)
+	if err != nil {
+		return nil, fmt.Errorf("could not create test results: %v", err)
+	}
+
```

Comment:
I understand this is here to fail fast, but it would seem that if there are other failures (e.g. if `certImage.Repositories == 0` a few lines down), wouldn't we have created a TestResult for this account before we've asserted whether their content is valid?

Reply:
Good catch, that should probably be first, since it was lost in the comments, I'll move it to be first thing.